### PR TITLE
ci(cypress): Add Dynamic fields Test Cases

### PR DESCRIPTION
**Pull request by likhinbopanna**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## Description
Added Dynamic field Test cases  for cybersource connector 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Currently our automation setup doesn't have dynamic field test cases 

## How did you test it?

image

image

- cybersource
 
## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import apiKeyCreateBody from "../../fixtures/create-api-key-body.json";
 import createConnectorBody from "../../fixtures/create-connector-body.json";
 import merchantCreateBody from "../../fixtures/merchant-create-body.json";
+import customerCreateBody from "../../fixtures/create-customer-body.json";
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:4]: Why are we doing this when we clearly have `imports.js` which we're SUPPOSED to be using?

Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:4]: Everywhere fixtures.body is not being used , do you want me change everywhere?

Comment (pixincreate) [reply cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:4]: in payments and payouts, we're indeed using the import file:

```js
import * as fixtures from "../../fixtures/imports";`
...
... fixtures.<body_name> ...
```

so, yes, we should change it.

File: cypress-tests/cypress/e2e/PaymentUtils/Stripe.js [hunk 2]
```diff
@@ -944,4 +944,183 @@ export const connectorDetails = {
...
+                    required_field: "payment_method_data.card.card_cvc",
+                    display_name: "card_cvc",
+                    field_type: "user_card_cvc",
+                    value: null,
+                  },
+                  "payment_method_data.card.card_exp_month": {
+                    required_field: "payment_method_data.card.card_exp_month",
+                    display_name: "card_exp_month",
+                    field_type: "user_card_expiry_month",
+                    value: null,
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentUtils/Stripe.js:1116]: i feel these are just too repetitive. can we have these kind of data in commons directly?  

this is what i meant in my previous comment.

File: cypress-tests/cypress/e2e/PaymentTest/00026-DynamicFields.cy.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
+import * as fixtures from "../../fixtures/imports";
+import State from "../../utils/State";
+import { card_credit_enabled } from "../PaymentMethodListUtils/Commons";
```
[the remark this answers is not in the document]
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentTest/00026-DynamicFields.cy.js:3]: I have to make changes everywhere in the payment methods list tests where i have imported from

File: cypress-tests/cypress/e2e/PaymentUtils/Commons.js
[the remark this answers is not in the document]
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentUtils/Commons.js:1]: sure

**Review by pixincreate — APPROVED** [review]

**Review by Sakilmostak — APPROVED** [review]

**Pull request merged**